src/basic/pidref.c

```diff
@@ -14,7 +14,28 @@
 #include "signal-util.h"
 #include "stat-util.h"
 
-bool pidref_equal(const PidRef *a, const PidRef *b) {
+int pidref_acquire_pidfd_id(PidRef *pidref) {
+        assert(pidref);
+
+        if (!pidref_is_set(pidref))
+                return -ESRCH;
+
+        if (pidref->fd < 0)
+                return -ENOMEDIUM;
+
+        if (pidref->fd_id > 0)
+                return 0;
+
+        struct stat st;
+
+        if (fstat(pidref->fd, &st) < 0)
+                return -errno;
+
+        pidref->fd_id = st.st_ino;
```

Comment:
so, i'd prefer if we'd explicit check if the new inode numbers actually work and are supported. on old kernels the id is useless after all, will be the same for all processes. maybe have a helper somewhere like this:

```c
static int pidfd_inode_ids_supported(void) {
        _cleanup_close_ int fd = -EBADF;
        static int cached = -1;

        if (cached >= 0)
                return cached;

        fd = pidfd_open(getpid_cached(), 0);
        if (fd < 0) {
                if (ERRNO_IS_NOT_SUPPORTED(errno)) 
                        return (cached = false); 
                return -errno;
        }

        r = fd_is_fs_type(fd, PID_FS_MAGIC);
        if (r < 0)
                return r;

        return (cached = r);
}
```